### PR TITLE
[9.x] Improving the project structure

### PR DESCRIPTION
**Pull request by andrey-helldar**
Discussion: https://github.com/Laravel-Lang/lang/discussions/1606

### CODE REVIEW

File: app/main/Concerns/Contains.php [hunk 1]
```diff
@@ -8,11 +8,16 @@ trait Contains
 {
     protected function isJson(string $filename): bool
     {
-        return Str::endsWith($filename, '.json');
+        return Str::contains($filename, 'json');
```
Comment (caouecs) [inline app/main/Concerns/Contains.php:11]: Can we use [str_contains](https://www.php.net/manual/en/function.str-contains) directly ?

**Review by caouecs — COMMENTED** [review]
A question on Str::contains and PHP8 str_contains function

File: app/main/Concerns/Contains.php (hunk 1 above)
Comment (andrey-helldar) [reply app/main/Concerns/Contains.php:11]: Exactly! Forgot about this feature. Thank you!

Yes, we can use it. My package contains this feature for earlier php versions.

I'll update the code in a few minutes.

**Comment by caouecs** [comment]
In TestCase.php
line 64 :  return Str::startsWith($filename, 'validation'); => return str_starts_with($filename, 'validation');

lines 29, 34: return Arr::ksort($content);   I don't find it in [Laravel doc](https://laravel.com/docs/8.x/helpers), and it's ksort(), no ?

**Comment by andrey-helldar** [comment]
> In TestCase.php
> line 64 : return Str::startsWith($filename, 'validation'); => return str_starts_with($filename, 'validation');
> 
> lines 29, 34: return Arr::ksort($content); I don't find it in [Laravel doc](https://laravel.com/docs/8.x/helpers), and it's ksort(), no ?

`Str::startsWith` will be replaced

`Arr::ksort` - this method has never been in the framework. I wrote it myself for my own purposes and use it in several projects. See https://github.com/andrey-helldar/support/blob/main/src/Helpers/Arr.php#L143-L164

File: app/tests/TestCase.php [hunk 2]
```diff
@@ -61,7 +60,7 @@ protected function sort(array &$array): void
 
     protected function isValidation(string $filename): bool
     {
-        return Str::startsWith($filename, 'validation');
+        return str_contains($filename, 'validation');
```
Comment (caouecs) [inline app/tests/TestCase.php:63]: str_starts_with ?

**Review by caouecs — CHANGES_REQUESTED** [review]
Just to be sure

File: app/tests/TestCase.php (hunk 2 above)
Comment (andrey-helldar) [reply app/tests/TestCase.php]: Yeah sorry. My mistake. I need to sleep.

Comment (andrey-helldar) [reply app/tests/TestCase.php]: The code checks the occurrence of words on the file name, so you can use both the `str_starts_with` function and `str_contains`. Taking into account our implementation, changing them is tantamount to saving on matches. We can use both, but I prefer to write correctly right away. The file names `validation.php` and `validation-inline.php` always start with the `validation` word, so it is better to use the `str_starts_with` function.

**Comment by caouecs** [comment]
It's a major version, I'll merge tomorrow, the time to do all versions

**Comment by andrey-helldar** [comment]
https://github.com/andrey-helldar/laravel-lang-publisher/pull/84 is ready to work with version 9 and new structure 😊

**Comment by andrey-helldar** [comment]
Go go go))

**Comment by caouecs** [comment]
done, but we need to update pull requests :)

**Comment by andrey-helldar** [comment]
Yes, I wrote in them that the files were changed.

**Pull request merged**